ecl/hql/hqlgram2.cpp

```diff
@@ -12234,13 +12233,73 @@ void parseAttribute(IHqlScope * scope, IFileContents * contents, HqlLookupContex
         throw;
     }
     OwnedHqlExpr parsed = scope->lookupSymbol(name, LSFsharedOK|LSFnoreport, ctx);
-    OwnedHqlExpr simplified = createSimplifiedDefinition(parsed);
+    OwnedHqlExpr simplified;
+    if (ctx.syntaxChecking() || ctx.hasCacheLocation())
+    {
+        simplified.setown(createSimplifiedDefinition(parsed));
+
+        if (simplified && ctx.checkSimpleDef())
+            verifySimpifiedDefinition(simplified, attrCtx);
+    }
     bool canCache = parsed && (parsed->getOperator() != no_forwardscope);
     bool isMacro = parsed && parsed->isMacro();
 
     attrCtx.noteEndAttribute(true, canCache, isMacro, simplified);
+    if (simplified && (simplified != parsed) && ctx.syntaxChecking())
+        scope->defineSymbol(LINK(simplified));
+}
+
+IHqlExpression * parseDefinition(const char * ecl, IIdAtom * name, MultiErrorReceiver &errors)
```

Comment:
new functions need a comment to describe what they do - e.g., why does this differ from parseAttribute